Rest Client not using system calls

rmescandon opened the pull request:
Tested on armhf (rpi3)
Verified operations:
- set ssid
- set passphrase
- show config
- ap on
- ap off

Removed not needed yet unixhttpc tool

knitzsche commented:
Thanks Roberto. Couple  items:

* Do we still need wifiap.go? It now just calls your new DefaultRestClient functions one-to-one, like this:
func Show() {
        result, err := DefaultRestClient().Show()
func Enable() {
        err := DefaultRestClient().Enable()
...
So can't we delete wifiap.go totally with the understanding that any code that needs the rest api functions will simply call your functions instead? (If so, your Show() function would have to print the resulting map, not return it.)

If we don't drop wifiap.go, then every time we need to add a new rest api call we have to implement it twice.

* SNAP_COMMON/sockets
master branch uses $SNAP_COMMON/sockets to mount the wifi-ap:control shared content. But your code seems to refer to "/var/snap/wifi-ap/current/sockets/control", which is $SNAP_DATA/... I think:
https://github.com/CanonicalLtd/UCWifiConnect/pull/6/files#diff-537a7a9194f23d96fe38398b08177180R29

I'd prefer SNAP_COMMON because it persists through snap version increments, whereas SNAP_DATA does not, I think, unless there's a reason for this change. 

* I'd rather not output "Ok." after successful command execution, which is added in several places, such as:
https://github.com/CanonicalLtd/UCWifiConnect/pull/6/files#diff-537a7a9194f23d96fe38398b08177180R121

Do you agree it is reasonable to only output text if there is an error, or if the command clearly should output something (like show the ap config)?

* some of the diff is mere whitespace difference (I've been developing on pi3 and its default whitespace auto correction is not the same as desktop, so I don't get aligned structs for example). I prefer your better whitespace, not sure how to enable it in vim on the pi3 - tips?)

rmescandon commented:
1) I've moved the REST operations to wifiap.go, leaving in restclient.go only the generic methods used for any request. This way makes sense wifiap existence
2) Completely agree. For some (unknown) reason, when I made some tests I was only able to access remote snap folder path. But I've now tried original one in common folder and goes ok. Changed to SNAP_COMMON/sockets
3) TBH, I also was not too convinced of returning anything. Removed.
4) This is the only point to see how to solve. I used sublime text except for this last change, where I've made changes using Visual Studio Code. In both IDEs there are plugins formatting code. Let me know if now you see same lack of alignment in vi and we can discuss a good solution.

arapulido commented:
About formatting Go. There are no problems with Go and formatting. All plugins should be using goftm, which has very strict rules (always tabs!). 

 @knitzsche Are you using the vi Go plugin? (https://github.com/fatih/vim-go) If not, you should. That will format your code in the "Go way" :smile:
